test/app/model/ChessEngine.spec.ts

```diff
@@ -445,3 +444,187 @@ describe('Return player moves without those that cause his king to check', () =>
     );
   });
 });
+
+describe('Castling can be done only if neither king nor rook has moved and none of square which the king is passing by is under attack and no piece between the king and the rook', () => {
+  const whiteKing = new King(Side.WHITE);
+  const whiteRookA1 = new Rook(Side.WHITE);
+  const whiteRookH1 = new Rook(Side.WHITE);
+  const whiteBishop = new Bishop(Side.WHITE);
+  const blackKing = new King(Side.BLACK);
+  const blackRook = new Rook(Side.BLACK);
+  const blackKnight = new Knight(Side.BLACK);
+  const blackPawn = new Pawn(Side.BLACK);
+
+  it('When all conditions are fulfilled, then short castling can be done', () => {
+    const boardWithPieces: SquareWithPiece = { E1: whiteKing, H1: whiteRookH1 };
+    const chessBoard = new Chessboard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const kingSquareFrom: Square = { column: 'E', row: 1 };
+    const kingSquareTo: Square = { column: 'G', row: 1 };
+    const rookSquareFrom: Square = { column: 'H', row: 1 };
+    const rookSquareTo: Square = { column: 'F', row: 1 };
+
+    expect(engine.move(kingSquareFrom, kingSquareTo)).toIncludeSameMembers([
+      { eventType: 'PieceWasMoved', piece: whiteKing, from: kingSquareFrom, to: kingSquareTo },
+      { eventType: 'PieceWasMoved', piece: whiteRookH1, from: rookSquareFrom, to: rookSquareTo },
+    ]);
+  });
+
+  it('When all conditions are fulfilled, then long castling can be done', () => {
+    const boardWithPieces: SquareWithPiece = { E1: whiteKing, H1: whiteRookH1 };
+    const chessBoard = new Chessboard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const kingSquareFrom: Square = { column: 'E', row: 1 };
+    const kingSquareTo: Square = { column: 'C', row: 1 };
+    const rookSquareFrom: Square = { column: 'A', row: 1 };
+    const rookSquareTo: Square = { column: 'D', row: 1 };
+
+    expect(engine.move(kingSquareFrom, kingSquareTo)).toIncludeSameMembers([
+      { eventType: 'PieceWasMoved', piece: whiteKing, from: kingSquareFrom, to: kingSquareTo },
+      { eventType: 'PieceWasMoved', piece: whiteRookH1, from: rookSquareFrom, to: rookSquareTo },
+    ]);
```

Comment:
fajnie

Reply:
Jak najbardziej o to chodzilo :) Latwo to potem na UI ogarnac - w sumie to nic nie bedzie trzeba zmieniac - przestawiajac dwie bierki.